clang/test/SemaCXX/GH102064.cpp

```diff
@@ -0,0 +1,3 @@
+// RUN: %clang_cc1 -std=c++20 -fms-extensions %s
+// expected-no-diagnostics
+constexpr int x = []{ __noop; return 0; }();
```

Comment:
Can you move this test to test/SemaCXX/builtins.cpp ? thanks!

Reply:
@cor3ntin 
I have added this test to builtins.cpp.
I'm not sure if this is a good way to add this test.
Please have a look. Thanks!

Reply:
You do not need the define/ifdef/etc

Reply:
If adding ``RUN: %clang_cc1 %s -std=c++20 -fms-extensions`` at the top line of builtins.cpp, without using define/ifdef, other testcases will fail.
Could @cor3ntin kindly give me more hints to this?

Reply:
oh, because it's a MSVC extensions you can write 

```
#ifdef _MSC_VER
constexpr int x = []{ __noop; return 0; }();
#endif
```
(clang defines _MSC_VER, you do not need to do it)

Sorry, i did not realize it was a MS only extension